source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -141,6 +143,9 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(NextResultPtr&& next_result
     // used in tsi_socket_test.cc implements the interface returning the max frame size.
     tsi_zero_copy_grpc_protector_max_frame_size(frame_protector, &actual_frame_size_to_use_);
 
+    // Reset the watermarks with actual negotiated max frame size.
+    read_buffer_.setWatermarks(
+        std::max<size_t>(actual_frame_size_to_use_, callbacks_->connection().bufferLimit()));
```

Comment:
actual_frame_size_to_use_ + frame_overhead

Reply:
Not needed per previous comment.